src/vmspawn/vmspawn.c

```diff
@@ -920,6 +941,26 @@ static int run_virtual_machine(void) {
                         return log_oom();
         }
 
+        if (strv_isempty(arg_initrds) && arg_linux) {
+                KernelImageType kt;
+                r = inspect_kernel(AT_FDCWD, arg_linux, &kt, NULL, NULL, NULL);
```

Comment:
as discussed elsewhere, please prefix NULL arguments with comments with the argument names